src/trig.jl

```diff
@@ -20,9 +20,11 @@ positive and negative q) and if this condition is true we flip the sign since
 we are now in the negative branch of sin(x). Recall that q is just the integer
 part of d/π and thus we can determine the correct sign using this information.
 """
-@inline _sincos(x::Float64) = @horner(x, c1, c2, c3, c4, c5, c6, c7, c8, c9)
+@inline _sincos(x::Float64) = @horner x c1 c2 c3 c4 c5 c6 c7 c8 c9
 
 function xsin(d::Float64)
+    sgn = sign(d)
```

Comment:
I don't fully understand the Julia code, but doesn't the modification add one multiplication?
For changing the sign of an FP value, mulsign() should be faster than copysign.
BTW, why don't you use the original sin kernel?
sincos kernel should be slower.